src/sagas/helpers.js

```diff
@@ -6,12 +6,38 @@
  */
 
 import { get } from 'lodash';
-import { put, race, take, call } from 'redux-saga/effects';
+import {
+  put,
+  race,
+  take,
+  call,
+  select,
+} from 'redux-saga/effects';
 import { t } from 'ttag';
 import NavigationService from '../NavigationService';
 import {
-  setIsShowingPinScreen, types,
+  setIsShowingPinScreen,
+  types,
 } from '../actions';
+import { FEATURE_TOGGLE_DEFAULTS } from '../constants';
+
+export function* waitForFeatureToggleInitialization() {
+  const featureTogglesInitialized = yield select((state) => state.featureTogglesInitialized);
+
+  if (!featureTogglesInitialized) {
+    // Wait until featureToggle saga completed initialization, which includes
+    // downloading the current toggle status for this client.
+    yield take(types.FEATURE_TOGGLE_INITIALIZED);
+  }
+}
+
+export function* checkForFeatureFlag(flag) {
+  yield call(waitForFeatureToggleInitialization);
+
+  const featureToggles = yield select((state) => state.featureToggles);
```

Comment:
Why use `state.featureToggles` if we can access the `unleashClient` and ask `isEnabled('feature-toggle')` to it?

Reply:
We could, I prefer abstracting it and keeping everything that is unleash-related inside the saga, so we can easily migrate to another feature flag provider later on

What do you think?

Reply:
Mmmm... Ok. No problem.